torch/csrc/jit/passes/insert_guards.cpp

```diff
@@ -33,14 +33,20 @@ struct GuardInserter {
         // n->input() is Tensor type
         auto guard = graph_->create(prim::Guard, {n->input()}, 1);
         auto go = guard->output();
-        // make a *copy* of ProfilingTensorType, in case we'd like
-        // to make changes to it independently from the one being
-        // profiled
-        auto copy = ProfiledTensorType::create(
-            n->output()->type()->expect<ProfiledTensorType>());
-        go->setType(copy);
-        guard->insertBefore(n);
-        n->output()->replaceAllUsesWith(go);
+        auto pttp = n->output()->type()->cast<ProfiledTensorType>();
+        if (pttp) {
+          // make a *copy* of ProfilingTensorType, in case we'd like
+          // to make changes to it independently from the one being
+          // profiled
+          auto copy = ProfiledTensorType::create(pttp);
+          go->setType(copy);
+          guard->insertBefore(n);
+          n->output()->replaceAllUsesWith(go);
+        } else {
+          // we didn't go down this path i.e
+          // no profiling information is available
+          n->output()->replaceAllUsesWith(n->input());
```

Comment:
Where do we insert the guard in `else` condition?

Reply:
we don't 😄  ... the if arm this `prim::profile` is in wasn't executed, so we don't have any profiling information to guard